h/static/scripts/group-forms/index.tsx

```diff
@@ -3,7 +3,9 @@ import { render } from 'preact';
 import CreateGroupForm from './components/CreateGroupForm';
 
 function init() {
-  render(<CreateGroupForm />, document.querySelector('#create-group-form')!);
+  const shadowHost = document.querySelector('#create-group-form')!;
+  const shadowRoot = shadowHost.attachShadow({ mode: 'closed' });
```

Comment:
```suggestion
  const shadowRoot = shadowHost.attachShadow({ mode: 'open' });
```

I would always suggest using the `open` shadow root mode unless you really need to hide the internals from the surrounding page. Browsers use the `closed` mode for built-in HTML elements for example. Style isolation still applies regardless of the mode.